internal/manifest/manifest.go

```diff
@@ -180,3 +245,100 @@ func iteratePrefix[T prefixable](manifest *Manifest, prefix *T, onMatch func(*T)
 	}
 	return nil
 }
+
+func manifestAddPackages(dbw *jsonwall.DBWriter, infos []*archive.PackageInfo) error {
+	for _, info := range infos {
+		err := dbw.Add(&Package{
+			Kind:    "package",
+			Name:    info.Name,
+			Version: info.Version,
+			Digest:  info.SHA256,
+			Arch:    info.Arch,
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func manifestAddSlices(dbw *jsonwall.DBWriter, slices []*setup.Slice) error {
+	for _, slice := range slices {
+		err := dbw.Add(&Slice{
+			Kind: "slice",
+			Name: slice.String(),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func manifestAddReport(dbw *jsonwall.DBWriter, entries map[string]ReportEntry) error {
+	for _, entry := range entries {
+		sliceNames := []string{}
+		for slice := range entry.Slices {
+			err := dbw.Add(&Content{
+				Kind:  "content",
+				Slice: slice.String(),
+				Path:  entry.Path,
+			})
+			if err != nil {
+				return err
+			}
+			sliceNames = append(sliceNames, slice.String())
+		}
+		sort.Strings(sliceNames)
+		err := dbw.Add(&Path{
+			Kind:      "path",
+			Path:      entry.Path,
+			Mode:      fmt.Sprintf("0%o", unixPerm(entry.Mode)),
+			Slices:    sliceNames,
+			Hash:      entry.Hash,
+			FinalHash: entry.FinalHash,
```

Comment:
These should be named after the respective algorithm instead of being Hash and FinalHash. Also, I think the standard is Sha256 instead of SHA256, but please have a look at the standard library to see what's the common convention.

It's okay to address that in a follow up if you'd prefer that.

Reply:
Done in https://github.com/canonical/chisel/pull/142/commits/86c6e7be7b4c6f0b9193fb5bcc3812d851af5b99, I changed all the made sense in the chain.